docs/proposals/service-discovery/README.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Service discovery with native Kubernetes naming and resolution
+authors:
+- "@bivas"
+reviewers:
+- "@XiShanYongYe-Chang"
+- TBD
+approvers:
+- "@robot"
+- TBD
+
+creation-date: 2023-06-22
+
+---
+
+# Service discovery with native Kubernetes naming and resolution
+
+## Summary
+
+With the current `ServiceImportController` when a `ServiceImport` object is reconciled, the derived service is prefixed with `derived-` prefix.
+This seems counterintuitive when thinking about service discovery:
+- Assuming the pod is exported as the service `foo`
+- Another pod that wishes to access it on the same cluster will simply call `foo` and Kubernetes will bind to the correct one
+- If that pod is scheduled to another cluster, the original service discovery will fail as there's no service by the name `foo`
+- To find the original pod, the other pod is required to know it is in another cluster and use `derived-foo` to work properly
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.
+-->
+
+### Goals
+
+- Remove the "derived-" prefix from the service
+- User-friendly and native service discovery
+
+### Non-Goals
+
+- Multi cluster connectivity
+
+## Proposal
+
+Following are flows to support the service import proposal:
+
+1. `Deployment` and `Service` are created on cluster member1 and the `Service` imported to cluster member2 using `ServiceImport` (described below as [user story 1](#story-1))
+2. `Deployment` and `Service` are created on cluster member1 and both propagated to cluster member2. `Service` from cluster member1 is imported to cluster member2 using `ServiceImport` (described below as [user story 2](#story-2))
+
+The proposal for this flow is what can be referred to as local-and-remote service discovery. The options could be:
+
+1. **Local** only - In case there's a local service by the name `foo` Karmada never attempts to import the remote service and doesn't create an `EndPointSlice`
+2. **Local** and **Remote** - Users accessing the `foo` service will reach either member1 or member2
+3. **Remote** only - in case there's a local service by the name `foo` Karmada will remove the local `EndPointSlice` and will create an `EndPointSlice` pointing to the other cluster (e.g. instead of resolving member2 cluster is will reach member1)
```

Comment:
`Remote` under what circumstances would this be used? I feel like we could consider temporarily not supporting this method.

Reply:
`Remote` can be used for cases like privacy or business consideration that doesn't allow processing of customer data in specific location and we must pass the traffic to privacy controlled cluster.

Reply:
For users, if they want to perform remote processing, why do they need to deploy a deployment in the current cluster? They can simply deploy it in another cluster.

Furthermore, from an implementation perspective, how can we ensure the deletion of EndpointSlice? This is controlled by controllers in the cluster, so even if we delete it, a new one will be created immediately.

Reply:
I agree it is a rare case. I can think of a monolith that has multiple entry points and it can read messages of Kafka broker but isn't allowed to get REST API calls. So the user will deploy the code using a `Deployment` but will not want it exposed as REST endpoint. Faulty user architecture, but can work with Karmada. 

I don't have a good idea how to delete it, maybe we can push that feature for later.

Reply:
Thanks for your response!

Reply:
I basically agree with @XiShanYongYe-Chang that we might not touch the `local EndpointSlice` which is managed by the member cluster. 

I understand the case for `Remote Only` should be that:
There are no Pods on that cluster, but the cluster needs to consume the service. 
For example, a user might deploy DB services on a cluster, and allow other clusters to access the service.